shears/mine_for_gold.py

```diff
@@ -36,23 +42,57 @@ def main():
     args = parser.parse_args()
 
     print('Authenticating Reddit API connection...')
+    
+    """Authorized Reddit Instances"""
     reddit = praw.Reddit(
         client_id=args.client_id,
         client_secret=args.client_secret,
         username=_USERNAME,
         password=args.password,
         user_agent=_USER_AGENT,
     )
+
+    """ To verify that you are authenticated as the correct user """
+    print(reddit.user.me())     
+    
     print('Authenticated!')
 
+
     print('Mining for gold...')
     whitelist = get_ticker_set()
-    for comment in reddit.subreddit('wallstreetbets').comments():
-        print(comment.author.name + ' says:\n')
-        print(comment.body)
-        print('Ticker counts: ' + repr(scrape_tickers(comment.body,
-                                                      whitelist=whitelist)))
 
+    subredditName = "wallstreetbets"
+
+    updated_json = {}
+    updated_json[subredditName] = []
+
+    try:
+        with open('comments.json') as current_comments: 
+            try:   
+                current_data = json.load(current_comments)
+            except : 
+                current_data = {}
+            new_and_old_comments = []
+            if subredditName in current_data:
+                for dicObj in current_data[subredditName]:
+                    new_and_old_comments.append(dicObj)
```

Comment:
This can be simplified to

```python
new_and_old_comments.extend(current_data[subredditName])
```

The official Python docs are a great way to find convenient methods like these: https://docs.python.org/3/tutorial/datastructures.html#more-on-lists